### DISCUSSION TITLE
How to create a white channel?

### DISCUSSION THREAD

**Discussion by iwhp** [dlemstra/Magick.NET#1394] [Help]
I try to create a white channel, but I cannot figure out how to do that.

If there is an image which contains a rectangle of another image and some text next to the image, the area where the image should be a solid rectangle in the white channel and the text should also be outlined in the white channel.

Any ideas of how to accomplish this? Many thankx!

**Comment by dlemstra** [comment]
I am sorry but I have no clue what you are trying to ask me?

**Reply by iwhp** [reply]
I am sorry. Lets try it this way: we need to print an image on a bottle. To make sure that everything looks good, we use the white color as the base color on everything that gets printed. So the white color gets printed first, then comes the image itself. If there is an area in the image, which is an image by itself, then there will be a white rectangle printed first, then on that surface the image is printed. If there is a text, we first print the text in white, then we print the surface in e.g. red. We print in cymk, so we need to add a channel called White and this contains all the printable parts of the cymk channels. Does that make sense now?

**Reply by dlemstra** [reply]
I am sorry but I still don't really understand what you are trying to do. It sounds like you need a specific output format? What kind of format do you need?

**Reply by iwhp** [reply]
Hi Dirk. Thankx for coming back!

I need a tif/CMYK file. In the color channels it will have a C, M, Y, K and White channel.
So the first challenge I am facing is how to create this White channel. It should be named "White". Is this possible to add a 5 channel and name it "White"?
The second challenge I am facing is to get the colors from C, M, Y and K channels all combined in this White channel. It will be colored to a single color: let it be red. 

This white channel will then be interpreted by the printer to print the white color. On top of this white color, all other colors are printed.

**Reply by dlemstra** [reply]
A tiff file can also "secretly" contain psd layer information. Is it possible to share a file that does what you want to do? It might be possible to do it with this library but I will need a small simple example to investigate that.

**Reply by iwhp** [reply]
Thankx, please find the sample in this zip file: [sample.zip](https://github.com/dlemstra/Magick.NET/files/11673475/sample.zip)
